resources/grabs.js

```diff
@@ -0,0 +1,128 @@
+const axios = require('axios');
+const db = require('../server/db/index.js');
+const fs = require('fs');
+const path = require('path');
+
+// Make sure our 'courseLists' directory is available to use
+if ( ! fs.existsSync(path.join(__dirname, 'courseLists'))) {
+  fs.mkdirSync(path.join(__dirname, 'courseLists'), 0777);
+}
+
+const currentFiles = fs.readdirSync(path.join(__dirname, 'courseLists'));
+
+// Add all files to create here as needed
+const filesToCheck = ["khanAcademy"];
+// Add to apiInformation as needed
+const apiInformation = [
+  { fileName: 'khanAcademy',
+    info: { providerName: 'Khan Academy',
+	    url: 'http://www.khanacademy.org/api/v1/topictree?kind=topic',
+	    headers: '',
+	    func: createKhanCourse,
+	    apikey: 'unknown',
+	  },
+    func: createKhanCourse,
+  }
+];
+
+const currentTime = new Date();
+// One week time in milliseconds - So not really a magic number anymore
+const weekMS = 604800000;
+
+const listOfGets = filesToCheck.filter(file => {
+  if(!currentFiles.includes(file)) return true;
+  const stats = fs.statSync(path.join(__dirname, 'courseLists', file));
+  // Check and see if the existing file is more than a week old
+  // If it is we add to the list to get new data
+  if(currentTime - stats.mtime > weekMS) return true;
+  return false;
+});
+
+getAllCourses();
+
+async function getAllCourses() {
+  // Once we have a seed file this can happen there
+  // Presumably we wont want to sync if calling this periodically on the server
+  await db.db.sync({ force: true });
```

Comment:
why is it db.db.sync?

Reply:
Its funny because @justin-cook-developer asked the same question. 
Its because of the way we are exporting everything in our db index.js file. One of the things exported is db, and of course when I import it into this file I just called it db as well. So drilling into the namespace db, I have the db object. :smile:

Reply:
haha, oh okay I gotcha. I was a bit confused